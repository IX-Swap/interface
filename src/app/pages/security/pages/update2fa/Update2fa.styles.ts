import { makeStyles } from '@material-ui/core/styles'

export const useStyles = makeStyles(theme => ({
  wrapper: {
    backgroundColor: theme.palette.background.paper
  }
}))
