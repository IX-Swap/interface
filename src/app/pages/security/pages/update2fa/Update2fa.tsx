import React, { useEffect, useState } from 'react'
import { Container, Box, Stepper, Step, StepLabel, Grid } from '@mui/material'
import { TwoFaData } from 'app/pages/security/types'
import { ActiveStep } from 'app/pages/security/pages/update2fa/components/ActiveStep'
import { PageHeader } from 'app/components/PageHeader/PageHeader'
import { useStyles } from './Update2fa.styles'
import { ChangeStepButtons } from 'app/pages/security/components/ChangeStepButtons'
import { useAuth } from 'hooks/auth/useAuth'
import { history } from 'config/history'
import { SecurityRoute } from 'app/pages/security/router/config'
import { VSpacer } from 'components/VSpacer'

const steps = [
  'Remove Current Authenticator',
  'Scan New QR Code',
  'Backup the Key',
  'Enable New Authenticator'
]

export const Update2fa = () => {
  const classes = useStyles()
  const [activeStep, setActiveStep] = useState(0)
  const [twoFaData, setTwoFaData] = useState<TwoFaData | undefined>(undefined)
  const { user } = useAuth()

  useEffect(() => {
    if (user !== undefined && !user.totpConfirmed) {
      history.push(SecurityRoute.setup2fa)
    }
  })
  const nextStep = () => {
    setActiveStep(activeStep + 1)
  }
  const prevStep = () => {
    setActiveStep(activeStep - 1)
  }

  const handleSuccessfulFirstStep = (twoFaData: TwoFaData) => {
    setTwoFaData(twoFaData)
    nextStep()
  }

  const isBackButtonVisible = activeStep > 1 && activeStep < steps.length
  const isNextButtonVisible = activeStep < steps.length - 1 && activeStep > 0

  return (
    <Grid container spacing={0}>
      <PageHeader title='Change Authenticator' />
      <Grid item xs={12} className={classes.wrapper}>
        <Container>
          <PageHeader title='Change Authenticator' variant={'h6'} />
          <Box>
            <Stepper activeStep={activeStep} alternativeLabel>
              {steps.map(label => (
                <Step key={label}>
                  <StepLabel>{label}</StepLabel>
                </Step>
              ))}
            </Stepper>
            <Grid
              container
              justifyContent='center'
              alignItems='center'
              direction='column'
            >
              <Grid item>
                <VSpacer size={'medium'} />
                <Box mt={4} mb={6} width='100%'>
                  <ActiveStep
                    twoFaData={twoFaData}
                    handleSuccessfulRemoveAuthenticator={
                      handleSuccessfulFirstStep
                    }
                    index={activeStep}
                    nextStep={nextStep}
                  />
                </Box>
              </Grid>
              <Grid item>
                <ChangeStepButtons
                  isBackButtonVisible={isBackButtonVisible}
                  isNextButtonVisible={isNextButtonVisible}
                  onBackButtonClick={prevStep}
                  onNextButtonClick={nextStep}
                />
              </Grid>
            </Grid>
          </Box>
        </Container>
      </Grid>
    </Grid>
  )
}
