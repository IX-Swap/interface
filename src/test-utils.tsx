import React from 'react'
import { Router } from 'react-router-dom'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import {
  createGenerateClassName,
  StylesProvider
} from '@material-ui/core/styles'
import { ThemeProvider } from '@material-ui/styles'
import Themes from './v2/themes'
import { history } from 'v2/history'
import { UserProvider } from 'v2/auth/context'
import { UserStore } from 'v2/auth/context/store'
import { PasswordResetProvider } from 'v2/auth/context/password-reset'
import PasswordResetStore from 'v2/auth/context/password-reset/store'
import { PasswordResetStep } from 'v2/auth/context/password-reset/types'
import { AuthorizerTableStore } from 'v2/app/pages/authorizer/context/store'
import { AuthorizerTableStoreProvider } from 'v2/app/pages/authorizer/context'
import { DepositStore } from 'v2/app/pages/accounts/pages/banks/context/store'
import { DepositStoreProvider } from 'v2/app/pages/accounts/pages/banks/context'
import { ServicesProvider } from 'v2/services/useServices'
import { renderHook, RenderHookResult } from '@testing-library/react-hooks'
import { BreadcrumbsProvider } from 'v2/hooks/useBreadcrumbs'

const generateClassName = createGenerateClassName({
  productionPrefix: 'ix'
})

const BaseProviders: React.FC = ({ children }) => {
  return (
    <StylesProvider generateClassName={generateClassName}>
      <ThemeProvider theme={Themes.default}>
        <BreadcrumbsProvider>
          <Router history={history}>{children}</Router>
        </BreadcrumbsProvider>
      </ThemeProvider>
    </StylesProvider>
  )
}

const customRenderer = (
  ui: any,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult => {
  return render(ui, { wrapper: BaseProviders, ...options })
}

export const testComponentId = 'test-component'

export const fakeUserStore: Partial<UserStore> = {
  activeTab: 0,
  setActiveTab: jest.fn()
}

export const renderWithUserStore = (
  ui: any,
  store?: Partial<UserStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <UserProvider value={{ ...fakeUserStore, ...store }}>
        {children}
      </UserProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakePasswordResetStore: Partial<PasswordResetStore> = {
  currentStep: PasswordResetStep.Request,
  setCurrentStep: jest.fn()
}

export const renderWithPasswordResetStore = (
  ui: any,
  store?: Partial<PasswordResetStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <PasswordResetProvider value={{ ...fakePasswordResetStore, ...store }}>
        {children}
      </PasswordResetProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakeAuthorizerTableStore: Partial<AuthorizerTableStore> = {
  idKey: '__',
  uri: '/api/source',
  _getItemId: jest.fn()
}

export const renderWithAuthorizerTableStore = (
  ui: any,
  store?: Partial<AuthorizerTableStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <AuthorizerTableStoreProvider
        value={{ ...fakeAuthorizerTableStore, ...store }}
      >
        {children}
      </AuthorizerTableStoreProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakeDepositStore: Partial<DepositStore> = {}

export const renderWithDepositStore = (
  ui: any,
  store?: Partial<DepositStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <DepositStoreProvider value={{ ...fakeDepositStore, ...store }}>
        {children}
      </DepositStoreProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const renderHookWithServiceProvider = (
  hookFn: any,
  store: object = {}
): RenderHookResult<any, any> => {
  const WithServiceProvider: React.FC = ({ children }) => (
    <ServicesProvider value={store}>{children}</ServicesProvider>
  )

  return renderHook(hookFn, { wrapper: WithServiceProvider })
}

export * from '@testing-library/react'
export { customRenderer as render }
