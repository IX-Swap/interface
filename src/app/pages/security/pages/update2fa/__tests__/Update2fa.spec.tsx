import React from 'react'
import { render } from 'test-utils'
import { Update2fa } from 'app/pages/security/pages/update2fa/Update2fa'

describe('Update2fa', () => {
  it('does not render Next and Back button if step is 0, should match snapshot', () => {
    const { container, queryByText } = render(<Update2fa />)
    expect(queryByText('Next')).not.toBeInTheDocument()
    expect(queryByText('Back')).not.toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
