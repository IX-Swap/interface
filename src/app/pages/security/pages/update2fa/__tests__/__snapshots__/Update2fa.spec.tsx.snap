// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update2fa does not render Next and Back button if step is 0, should match snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container makeStyles-wrapper-1 css-1lym95h-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item makeStyles-leftBlock-2 css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-2xfiep-MuiTypography-root"
          >
            Update Authenticator by 
            <br />
             Verifying Your Identity
          </h2>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-t0zib5-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter makeStyles-description-7 css-1gd5lqg-MuiTypography-root"
              >
                To update your Authenticator please remove the current one
              </p>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item makeStyles-info-5 css-13i4rnv-MuiGrid-root"
            >
              <div
                class="makeStyles-wrapper-8 MuiBox-root css-0"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 makeStyles-text-9 css-1b4l2pg-MuiTypography-root"
                >
                  Click send to get a code on your email
                </p>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item makeStyles-formBlock-6 css-13i4rnv-MuiGrid-root"
            >
              <form
                data-testid="remove-authenticator-form"
                style="width: 100%;"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-80wlp8-MuiGrid-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root makeStyles-root-13 css-octvoh-MuiFormControl-root-MuiTextField-root"
                        control="[object Object]"
                        path="emailCode"
                      >
                        <label
                          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-hlid6-MuiFormLabel-root-MuiInputLabel-root"
                          data-shrink="true"
                          for="emailCode"
                          id="emailCode-label"
                        >
                          Verification Code
                        </label>
                        <div
                          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-1jk0dkh-MuiInputBase-root-MuiOutlinedInput-root"
                        >
                          <input
                            aria-invalid="false"
                            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1vvkud8-MuiInputBase-input-MuiOutlinedInput-input"
                            id="emailCode"
                            name="emailCode"
                            placeholder="E-mail verification code"
                            type="text"
                            value=""
                          />
                          <p
                            class="MuiTypography-root MuiTypography-body1 makeStyles-wrapper-16 css-1b4l2pg-MuiTypography-root"
                          >
                            Send
                          </p>
                          <fieldset
                            aria-hidden="true"
                            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="css-173wfxe"
                            >
                              <span>
                                Verification Code
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 makeStyles-helperText-10 css-1b4l2pg-MuiTypography-root"
                      >
                        Enter the 6-digit code sent to 
                        
                      </p>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-80wlp8-MuiGrid-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 makeStyles-label-11 css-1b4l2pg-MuiTypography-root"
                      >
                        Authenticator Code
                      </p>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-container MuiGrid-item css-19hkhe-MuiGrid-root"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                      >
                        <div
                          class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
                        >
                          <div
                            class="makeStyles-container-18 makeStyles-fullwidth-19"
                            style="display: flex;"
                          >
                            <div
                              style="display: flex; align-items: center;"
                            >
                              <input
                                aria-label="Please enter verification code. Digit 1"
                                autocomplete="off"
                                class="makeStyles-input-21  "
                                maxlength="1"
                                placeholder="_"
                                style="width: 1em; text-align: center;"
                                type="tel"
                                value=""
                              />
                            </div>
                            <div
                              style="display: flex; align-items: center;"
                            >
                              <input
                                aria-label="Digit 2"
                                autocomplete="off"
                                class="makeStyles-input-21 "
                                maxlength="1"
                                placeholder="_"
                                style="width: 1em; text-align: center;"
                                type="tel"
                                value=""
                              />
                            </div>
                            <div
                              style="display: flex; align-items: center;"
                            >
                              <input
                                aria-label="Digit 3"
                                autocomplete="off"
                                class="makeStyles-input-21 "
                                maxlength="1"
                                placeholder="_"
                                style="width: 1em; text-align: center;"
                                type="tel"
                                value=""
                              />
                            </div>
                            <div
                              style="display: flex; align-items: center;"
                            >
                              <input
                                aria-label="Digit 4"
                                autocomplete="off"
                                class="makeStyles-input-21 "
                                maxlength="1"
                                placeholder="_"
                                style="width: 1em; text-align: center;"
                                type="tel"
                                value=""
                              />
                            </div>
                            <div
                              style="display: flex; align-items: center;"
                            >
                              <input
                                aria-label="Digit 5"
                                autocomplete="off"
                                class="makeStyles-input-21 "
                                maxlength="1"
                                placeholder="_"
                                style="width: 1em; text-align: center;"
                                type="tel"
                                value=""
                              />
                            </div>
                            <div
                              style="display: flex; align-items: center;"
                            >
                              <input
                                aria-label="Digit 6"
                                autocomplete="off"
                                class="makeStyles-input-21 "
                                maxlength="1"
                                placeholder="_"
                                style="width: 1em; text-align: center;"
                                type="tel"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 makeStyles-helperText-10 css-1b4l2pg-MuiTypography-root"
                      >
                        Enter the 6-digit code from your existing Authenticator App
                      </p>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                    >
                      <button
                        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-disableElevation MuiButtonBase-root Mui-disabled makeStyles-button-12 css-r7zh34-MuiButtonBase-root-MuiButton-root"
                        disabled=""
                        tabindex="-1"
                        type="submit"
                      >
                        Remove and Continue
                      </button>
                    </div>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item makeStyles-rightBlock-3 css-5x0lb4-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-f93uvp"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-93b3zb-MuiTypography-root"
          >
            Progress
          </p>
        </div>
        <div
          class="MuiStepper-root MuiStepper-vertical css-9tmuzo-MuiStepper-root"
        >
          <div
            class="MuiStep-root MuiStep-vertical css-sei0vp-MuiStep-root"
          >
            <button
              class="MuiButtonBase-root MuiStepButton-root MuiStepButton-vertical active css-rko9yk-MuiButtonBase-root-MuiStepButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiStepLabel-root MuiStepLabel-vertical css-1bl8ooz-MuiStepLabel-root"
              >
                <span
                  class="MuiStepLabel-iconContainer css-4m07z1-MuiStepLabel-iconContainer"
                >
                  <div
                    class="makeStyles-wrapper-24 makeStyles-wrapper-27 MuiBox-root css-0"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 makeStyles-text-26 makeStyles-text-29 css-1b4l2pg-MuiTypography-root"
                    >
                      1
                    </p>
                  </div>
                </span>
                <span
                  class="MuiStepLabel-labelContainer css-1n74cx2-MuiStepLabel-labelContainer"
                >
                  <span
                    class="MuiStepLabel-label Mui-active css-t1n505-MuiStepLabel-label"
                  >
                    Remove Authenticator
                  </span>
                </span>
              </span>
              <span
                class="MuiTouchRipple-root MuiStepButton-touchRipple css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiStep-root MuiStep-vertical css-sei0vp-MuiStep-root"
          >
            <button
              class="MuiButtonBase-root MuiStepButton-root MuiStepButton-vertical css-rko9yk-MuiButtonBase-root-MuiStepButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiStepLabel-root MuiStepLabel-vertical css-1bl8ooz-MuiStepLabel-root"
              >
                <span
                  class="MuiStepLabel-iconContainer css-4m07z1-MuiStepLabel-iconContainer"
                >
                  <div
                    class="makeStyles-wrapper-24 makeStyles-wrapper-30 MuiBox-root css-0"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 makeStyles-text-26 makeStyles-text-32 css-1b4l2pg-MuiTypography-root"
                    >
                      2
                    </p>
                  </div>
                </span>
                <span
                  class="MuiStepLabel-labelContainer css-1n74cx2-MuiStepLabel-labelContainer"
                >
                  <span
                    class="MuiStepLabel-label css-t1n505-MuiStepLabel-label"
                  >
                    Scan QR Code
                  </span>
                </span>
              </span>
              <span
                class="MuiTouchRipple-root MuiStepButton-touchRipple css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiStep-root MuiStep-vertical css-sei0vp-MuiStep-root"
          >
            <button
              class="MuiButtonBase-root MuiStepButton-root MuiStepButton-vertical css-rko9yk-MuiButtonBase-root-MuiStepButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiStepLabel-root MuiStepLabel-vertical css-1bl8ooz-MuiStepLabel-root"
              >
                <span
                  class="MuiStepLabel-iconContainer css-4m07z1-MuiStepLabel-iconContainer"
                >
                  <div
                    class="makeStyles-wrapper-24 makeStyles-wrapper-33 MuiBox-root css-0"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 makeStyles-text-26 makeStyles-text-35 css-1b4l2pg-MuiTypography-root"
                    >
                      3
                    </p>
                  </div>
                </span>
                <span
                  class="MuiStepLabel-labelContainer css-1n74cx2-MuiStepLabel-labelContainer"
                >
                  <span
                    class="MuiStepLabel-label css-t1n505-MuiStepLabel-label"
                  >
                    Backup Key
                  </span>
                </span>
              </span>
              <span
                class="MuiTouchRipple-root MuiStepButton-touchRipple css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiStep-root MuiStep-vertical css-sei0vp-MuiStep-root"
          >
            <button
              class="MuiButtonBase-root MuiStepButton-root MuiStepButton-vertical css-rko9yk-MuiButtonBase-root-MuiStepButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiStepLabel-root MuiStepLabel-vertical css-1bl8ooz-MuiStepLabel-root"
              >
                <span
                  class="MuiStepLabel-iconContainer css-4m07z1-MuiStepLabel-iconContainer"
                >
                  <div
                    class="makeStyles-wrapper-24 makeStyles-wrapper-36 MuiBox-root css-0"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 makeStyles-text-26 makeStyles-text-38 css-1b4l2pg-MuiTypography-root"
                    >
                      4
                    </p>
                  </div>
                </span>
                <span
                  class="MuiStepLabel-labelContainer css-1n74cx2-MuiStepLabel-labelContainer"
                >
                  <span
                    class="MuiStepLabel-label css-t1n505-MuiStepLabel-label"
                  >
                    Enable Authenticator
                  </span>
                </span>
              </span>
              <span
                class="MuiTouchRipple-root MuiStepButton-touchRipple css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container makeStyles-container-39 css-1bizp5f-MuiGrid-root"
  />
</div>
`;
